File: Curso Programação Orientada a Objetos/Java com POO/Aula11/src/teste/Professor.java
package teste;

public class Professor extends Pessoa {
    //Atributos
    private String especialidade;
    private float salario;
    
    //Métodos
    public void receberAumento(float x) {
        System.out.println("Aumento no salário do professor " + this.nome);
        this.setSalario(this.getSalario() + x);
        //this.salario += x;
        System.out.println("Aumento de " + x);
        System.out.println("Salário agora é " + this.getSalario());
    }
    
    //Getters e Setters
    //especialidade
    public String getEspecialidade() {
        return especialidade;
    }
    public void setEspecialidade(String especialidade) {
        this.especialidade = especialidade;
    }

    //salario
    public float getSalario() {
        return salario;
    }
    public void setSalario(float salario) {
        this.salario = salario;
    }
    
    //Método toString
    @Override
    public String toString() {
        return "Professor{" + "nome=" + nome + ", idade=" + idade + ", sexo=" + sexo + 
                " especialidade=" + especialidade + ", salario=" + salario + '}';
    }
    
    
}

File: Curso Programação Orientada a Objetos/PHP com POO/Aula10/Professor.php
<?php

require_once 'Pessoa.php';

class Professor extends Pessoa{
    //Atributos
    private $especialidade;
    private $salario;
    
    //Métodos
    public function receberAum($x) {
        $this->setSalario($this->getSalario() + $x);
        //$this->salario += $x;
    }
    
    //Getters
    public function getEspecialidade() {
        return $this->especialidade;
    }

    public function getSalario() {
        return $this->salario;
    }
    
    //Setters
    public function setEspecialidade($especialidade) {
        $this->especialidade = $especialidade;
    }

    public function setSalario($salario) {
        $this->salario = $salario;
    }


}
